overlays/name-and-version.nix

```
@@ -0,0 +1,22 @@
+final: prev:
+let
+  inherit (prev) lib;
+  inherit (import ../lib { inherit lib; }) checkMkDerivationFor;
+
+  checkDerivation = drvArgs: drv:
+    lib.singleton {
+      name = "name-and-version";
+      cond = drvArgs ? name
+          && drvArgs ? version
+          && !(drvArgs ? pname);
+      msg = ''
+        Both `name` and `version` arguments are set. Did you mean `pname` instead?
```

Maybe something like

```suggestion
        Did you mean to pass `pname` instead of `name` to `mkDerivation`?
```

or even something like

```nix
if lib.hasSuffix "-${drvArgs.version}" drvArgs.name then ''
  Both `name` and `version` arguments are passed. Using `pname` instead of `name` will allow you to not append the version suffix to `name` manually.
'' else ''
  Did you mean to pass `pname` instead of `name` to `mkDerivation`?
''
```

to distinguish the old style from a typo.